packages/design-system/src/scalars/components/date-picker-field/use-date-picker-field.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { DateFieldValue } from "./types";
+import { format, isValid } from "date-fns";
+
+interface DatePickerFieldProps {
+  value?: DateFieldValue;
+  defaultValue?: DateFieldValue;
+}
+
+export const useDatePickerField = ({
+  value,
+  defaultValue,
+}: DatePickerFieldProps) => {
+  const [date, setDate] = React.useState<Date>();
+  const [inputValue, setInputValue] = React.useState(
+    value ?? defaultValue ?? "",
+  );
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(e.target.value);
+  };
+
+  function handleDateSelect(date?: Date) {
+    setDate(date);
+    if (date) {
+      setInputValue(format(date, "MM/dd/yyyy"));
+    } else {
+      setInputValue("");
+    }
+  }
+  const formatDate = (value: DateFieldValue): string => {
+    if (!value) return "";
+
+    if (typeof value === "string") {
+      return value;
+    }
+
+    if (typeof value === "object" && isValid(value)) {
+      return format(value, "dd/MM/yyyy");
+    }
+    return "";
+  };
```

Comment:
you should arrow function or lambda function but no both styles or it could be confusing. I also recommend wrapping those functions in an `useCalback` as they're inside a hook or could be dependencies, this way we save some renders

Reply:
You are right done